### DISCUSSION TITLE
NIST - FIPS 203, 204 for CRYSTALS-Kyber and CRYSTALS-Dilithium

### DISCUSSION THREAD

**Discussion by wagner-robert** [openssl/openssl#25231] [General Discussion]
All, NIST approved FIPS 203 and 204 on Aug-13-2024 and just calls CRYSTALS-Kyber and CRYSTALS-Dilithium, ML-KEM and ML-DSA respectively. I was wondering if there are plans to update/and submit for FIPS validation in the future?
PS- Thanks for all of the great work that you do!!!

**Comment by tom-cosgrove-arm** [comment]
This question already seems to have been asked in #25184

**Reply by wagner-robert** [reply]
Not really, different questions.  They are asking if the algorithms will be integrated, I am really asking when it will be integrated,  finished and FIPS validated. I am trying to see if that is a 3,6,9,12,24,36 month process or what.  I am guessing this will happen in v3.x (has anyone plotted out the version)???  Since so many other products use OpenSSL as a base library - this seems to be the starting point in the push to PQC.  Any thoughts would be greatly appreciated.

**Reply by wagner-robert** [reply]
FYI - It appears https://csrc.nist.gov/Projects/cryptographic-module-validation-program/modules-in-process/modules-in-process-list OpenSSL has a FIPS module on hold since FIPS 140-3 | On Hold (4/25/2024)   April.  I wasn't sure if that one included PQC.

**Reply by paulidale** [reply]
That submission doesn't include any PQC support.
